astropy/utils/metadata/core.py

```diff
@@ -120,6 +126,7 @@ def __set__(self, instance, value):
             instance._meta = value
 
 
+@dataclass
```

Comment:
Is this actually useful here? It creates an `__init__`, a `__repr__` we don't use, and defines equality, which I'm not sure we want. I also replaces two lines with two others, adding indirection. Why not just keep it clean as it is?

Reply:
✅